compiler/ast.ml

```diff
@@ -16,41 +20,49 @@ type expr =
     | Literal_string of string
     | Literal_bool of bool
     | Id of string
-    | Binop of expr * bool_op * expr
     | Binop of expr * math_op * expr
-    | Call of string * actuals_opt
+    | Call of string * expr list
     (* Need to include array accessor here. -- Matt*)
 
-
 type arg_decl = {
     var_type   : string;
     var_name   : string;
 }
 
-type stmt =
-    | Expr of expr
-    | For of expr * bool_expr * assignment_stmt * stmt_list
-    | While of bool_expr * stmt_list
-    | Where of where_expr_list_opt * ID * stmt_list * where_lit
-    (* ID might need to be string, im not sure -- Gary *)
-    | If of bool_expr * stmt_list * stmt_list
-    | Return of expr
+type bool_expr =
+    | Binop of expr * bool_op * expr
+    | Bool_binop of bool_expr * conditional * bool_expr
     | Not of bool_expr
-    | Not of where_arg
-    | Assign of data_type * ID * expr
-    (* Same ID comment as above *)
 
-type where_expr_list =
-    | Where_cond of where_expr * And * where_expr
-    | Where_cond of where_expr * Or * where_expr
+type data_type =
+    | Int
+    | Float
+    | Bool
+    | String
+    | Json
+    | Array of data_type
+
+type assignment_stmt =
+    | Assign of data_type * string * expr
 
 type where_expr =
-    | Where_eval of where_arg * bool_op * where_arg
+    | Where_eval of expr * bool_op * expr
 
-type json_literal =
-    | Json_from_file of STRING_LITERAL
-    (* Same thing as ID from line 34, not sure if STRING_LITERAL
-                                should be just string -- Gary*)
+type where_expr_list =
+    | Where_cond of where_expr * conditional * where_expr
+
+type where_lit =
+    | Id of string
+    | Json_from_file of string
+
+type stmt =
+    | Expr of expr
```

Comment:
Those are the same thing.  I did it for consistency.

On Thu, Nov 12, 2015 at 2:20 PM, Anshul Gupta <notifications@github.com>
wrote:

> In compiler/ast.ml
> <https://github.com/mayankmahajan24/QL/pull/7#discussion_r44701499>:
>
> >
> > -type json_literal =
> > -    | Json_from_file of STRING_LITERAL
> > -    (* Same thing as ID from line 34, not sure if STRING_LITERAL
> > -                                should be just string -- Gary*)
> > +type where_expr_list =
> > +    | Where_cond of where_expr * conditional * where_expr
> > +
> > +type where_lit =
> > +    | Id of string
> > +    | Json_from_file of string
> > +
> > +type stmt =
> > +    | Expr of expr
>
> Should this line be "Expr of expr" instead of "| Expr of expr"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mayankmahajan24/QL/pull/7/files#r44701499>.
>